x/ibc/light-clients/solomachine/types/proof.go

```diff
@@ -0,0 +1,217 @@
+package types
+
+import (
+	"github.com/tendermint/tendermint/crypto"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	clientexported "github.com/cosmos/cosmos-sdk/x/ibc/02-client/exported"
+	clienttypes "github.com/cosmos/cosmos-sdk/x/ibc/02-client/types"
+	connectionexported "github.com/cosmos/cosmos-sdk/x/ibc/03-connection/exported"
+	connectiontypes "github.com/cosmos/cosmos-sdk/x/ibc/03-connection/types"
+	channelexported "github.com/cosmos/cosmos-sdk/x/ibc/04-channel/exported"
+	channeltypes "github.com/cosmos/cosmos-sdk/x/ibc/04-channel/types"
+	commitmenttypes "github.com/cosmos/cosmos-sdk/x/ibc/23-commitment/types"
+)
+
+// VerifySignature verifies if the the provided public key generated the signature
+// over the given data.
+func VerifySignature(pubKey crypto.PubKey, data, signature []byte) error {
+	if !pubKey.VerifySignature(data, signature) {
+		return ErrSignatureVerificationFailed
+	}
+
+	return nil
+}
+
+// EvidenceSignBytes returns the sign bytes for verification of misbehaviour.
+//
+// Format: {sequence}{data}
+func EvidenceSignBytes(sequence uint64, data []byte) []byte {
```

Comment:
agreed. But I'm not sure what a good replacement would be. I guess I could do `SignBytes`? but that feels a little too vague. Going to leave for now since it is only used by evidence